### DISCUSSION TITLE
Error feedback and Logging in Production

### DISCUSSION THREAD

**Discussion by tocaherge** [vercel/next.js#61572] [Help]
### Summary

I'm struggling with getting a decent setup for error handling and logging in App Router.

The obvious and documented way to handle it is through `error.tsx` files which works fine, but since the error provided to these gets stripped of any useful data for "security reasons" it's not possible to either display anything useful for the user or send anything useful to external services like Datadog.
The documentation has below example;

![Screenshot 2024-02-02 at 14 13 54](https://github.com/vercel/next.js/assets/153068127/ec1a5f4b-1a00-42fc-98b1-50802c044eed)

but this makes no sense to me, if the error is stripped of all information is production - what good does it do in an error reporting service?

Am I missing something obvious here? I would like to be able to opt out of the sensitive data scrubbing that Next does - I trust my users enough to not use the information maliciously...

**Comment by leerob** [comment]
Going to combine with this discussion, thank you https://github.com/vercel/next.js/discussions/62681.
